modules/installation-guide/partials/proc_importing-external-or-additional-tls-certificates.adoc

```diff
@@ -0,0 +1,218 @@
+----
+$ {orch-cli} rollout restart -n __<{project-context}-namespace-name>__ deployment/{prod-id-short}-operator
+$ {orch-cli} rollout restart -n __<{project-context}-namespace-name>__ deployment/{prod-id-short}/keycloak
+$ {orch-cli} rollout restart -n __<{project-context}-namespace-name>__ deployment/{prod-id-short}
+----
+
+== Verification 
+
+If you added the certificates without error, the {prod-short} server starts and obtains {identity-provider} configuration over HTTPS. Otherwise here is a list of things to verify: 
+
+- In case of a {prod-short} link:https://docs.openshift.com/container-platform/latest/operators/olm-what-operators-are.html[Operator] deployment, the `CheCluster` attribute `serverTrustStoreConfigMapName` value matches the name of the ConfigMap. Get the value using the following command :
++
+[subs="+attributes,+quotes",options="nowrap",role=white-space-pre]
+----
+$ {orch-cli} get -o json checluster/{prod-checluster} -n __<{project-context}-namespace-name>__ | jq .spec.server.serverTrustStoreConfigMapName
+----
++
+- {prod-short} Pod Volumes list contains one Volume that uses the ConfigMap as data-source. To get the list of Volumes of the {prod-short} Pod:
++
+[subs="+attributes,+quotes",options="nowrap",role=white-space-pre]
+----
+$ {orch-cli} get pod -o json __<{prod-id-short}-pod-name>__ -n __<{project-context}-namespace-name>__ | jq .spec.volumes
+----
++
+- {prod-short} mounts certificates in folder `/public-certs/` of the {prod-short} server container. This command returns the list of files in that folder:
++
+[subs="+attributes,+quotes",options="nowrap",role=white-space-pre]
+----
+$ {orch-cli} exec -t __<{prod-id-short}-pod-name>__ -n __<{project-context}-namespace-name>__ -- ls /public-certs/
+----
++
+- In the {prod-short} server logs there is a line for every certificate added to the Java truststore, including configured {prod-short} certificates.
++
+[subs="+attributes,+quotes",options="nowrap",role=white-space-pre]
+----
+$ {orch-cli} logs __<{prod-id-short}-pod-name>__ -n __<{project-context}-namespace-name>__
+(...)
+Found a custom cert. Adding it to java trust store based on /usr/lib/jvm/java-1.8.0/jre/lib/security/cacerts
+(...)
+----
++
+- ${prod-short} server Java trustore contains the certificates. The certificates SHA1 fingerpints are among the list of the SHA1 of the certificates included in the trustore returned by the following command:
++
+[subs="+attributes,+quotes",options="nowrap",role=white-space-pre]
+----
+$ {orch-cli} exec -t __<{prod-id-short}-pod-name>__ -n {prod-namespace} -- keytool -list -keystore /home/che/cacerts
+Your keystore contains 141 entries
+
+(...)
+----
++
+To get the SHA1 hash of a certificate on the local filesystem:
++
+[subs="+attributes,+quotes",options="nowrap",role=white-space-pre]
+----
+$ openssl x509 -in __<certificate-file-path>__ -fingerprint -noout
+SHA1 Fingerprint=3F:DA:BF:E7:A7:A7:90:62:CA:CF:C7:55:0E:1D:7D:05:16:7D:45:60
+----
```

Comment:
A verification at the workspace level: what config maps are created? what theia expects (an env variable? files mounted in a given folder? etc...)

Reply:
The problem is that as far as what I tested on CRW 2.4.0 release, the additional certificates added into the Che operator `ca-certs` config map **are not propagated to the workspace**: According to tests, and also according to [the code itself on the master branch](https://github.com/eclipse/che/blob/master/infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java#L81), the presence of a labelled `ca-certs` config map at the Che operator level, only triggers the creation of an **empty** `ca-certs` config map with the same annotation (used to inject OpenShift trusted CA bundles), but **without** the additional content that might have been manually added to the operator `ca-certs` config map.

So seems to completely invalidate the use of this config map to manage untrusted certificates that should be used also at the workspace level.

So maybe for now we should skip the workspace-related verification procedure and keep it for a future docs PR, when the propagation of the whole ca-certs CM content will be fixed.

@l0rd wdyt ?

Reply:
But at least trusted CA bundle certs are injected in the cm right? That should be mentioned and we could add instructions to verify it. Please open an issue and notify deployment/platform teams.

Reply:
Verification steps at the workspace have been added in commit https://github.com/eclipse/che-docs/pull/1598/commits/96614f0174a3a113190f330001c439683a17d8db